client/app/scripts/superdesk-dictionaries/dictionaries.js

```diff
@@ -63,7 +63,15 @@
         };
 
         this.update = function update(dictionary, data, success, error) {
-            return api.save('dictionaries', dictionary, data).then(success, error);
+            var sendData = {};
+            angular.forEach(data, function(val, key) {
+                if (key !== 'content') {
+                    sendData[key] = val === null? val: val.toString();
+                } else {
+                    sendData[key] = val;
+                }
```

Comment:
There are just two paths and I think that this is really readable. Indeed multiple imbricated "if" statements can become hard to read. But is not the case. And not always compact is better, it depends.